src/services/FrogbotService.ts

```diff
@@ -0,0 +1,166 @@
+            console.error('Error adding maintainers to environment:', error);
+        }
+    }
+
+    private async allowWorkflows(owner: string, repo: string) {
+
+        try {
+            await this.octokit.request(`PUT /repos/${owner}/${repo}/actions/permissions`, {
+                owner,
+                repo,
+                enabled: true,
+            });
+        } catch (error) {
+            throw error;
+        }
+    }
+
+    private async createBranch(owner: string, repo: string, defaultBranch: string) {
+        try {
+            const { data: refData } = await this.octokit.rest.git.getRef({
+                owner,
+                repo,
+                ref: `heads/${defaultBranch}`,
+            });
+
+            const latestCommitSha = refData.object.sha;
+            const newBranchRef = `refs/heads/${PULL_REQUEST_DATA.branchName}`;
+            await this.octokit.rest.git.createRef({
+                owner,
+                repo,
+                ref: newBranchRef,
+                sha: latestCommitSha,
+            });
+        } catch (error) {
+            console.error('Error creating branch:', error);
+        }
+    }
+
+    private async addFrogbotWorkflows(owner: string, repo: string, defaultBranch: string) {
+        try {
+            const scanRepoWorkflow = this.scanRepositoryWorkflow(defaultBranch);
+            const pullRequestWorkflow = this.pullRequestWorkflow();
+
+            await Promise.all([
+                this.octokit.rest.repos.createOrUpdateFileContents({
+                    owner,
+                    repo,
+                    path: '.github/workflows/frogbot-scan-repository.yml',
+                    message: `Added frogbot-scan-repository.yml on ${PULL_REQUEST_DATA.branchName}`,
+                    content: Buffer.from(scanRepoWorkflow).toString('base64'),
+                    branch: PULL_REQUEST_DATA.branchName,
+                }),
+                this.octokit.rest.repos.createOrUpdateFileContents({
+                    owner,
+                    repo,
+                    path: '.github/workflows/frogbot-scan-pull-request.yml',
+                    message: `Added frogbot-scan-pull-request.yml on ${PULL_REQUEST_DATA.branchName}`,
+                    content: Buffer.from(pullRequestWorkflow).toString('base64'),
+                    branch: PULL_REQUEST_DATA.branchName,
+                }),
+            ]);
+        } catch (error) {
+            console.error('Error adding workflows:', error);
+        }
+    }
+
+    private async openPullRequest(owner: string, repo: string, defaultBranch: string) {
+        await this.octokit.rest.pulls.create({
+            owner,
+            repo,
+            head: PULL_REQUEST_DATA.branchName,
+            base: defaultBranch,
+            title: PULL_REQUEST_DATA.prTitle,
+            body: PULL_REQUEST_DATA.comment,
+        });
+    }
+
+    private pullRequestWorkflow(): string {
+        return `Name: Frogbot pull request scan"
+jobs:
```

Comment:
```suggestion
        return `Name: Frogbot pull request scan"
on:
  pull_request_target:
    types: [opened, synchronize]
permissions:
  pull-requests: write
  contents: read
jobs:
```
missing some configs to the action